stdlib/public/core/AutoDiff.swift

```diff
@@ -495,6 +489,30 @@ public func gradient<T, U, V, R>(
   return { x, y, z in gradient(at: x, y, z, in: f) }
 }
 
+//===----------------------------------------------------------------------===//
+// `NoDerivative` attribute for properties
+//===----------------------------------------------------------------------===//
+
+@_propertyDelegate
+public struct NoDerivative<Value> {
+  public var value: Value
+
+  public init(_ value: Value) {
```

Comment:
Do you want this to be `init(initialValue:)` so that `@NoDerivative` properties can use the simple `=` initialization?

Reply:
Ah, yep.

Question: Why can't property delegates apply to `let` properties? This would be pretty crucial to autodiff use cases. For example:

```swift
struct MyVector {
   var x: Float
   var y: Float
   @NoDerivative let blah = Blah(...)
}
```